plugins/catalog-backend/src/processing/ProcessorOutputCollector.ts

```diff
@@ -137,7 +137,9 @@ export class ProcessorOutputCollector {
         },
       };
 
-      this.deferredEntities.push({ entity, locationKey: location });
+      const locationKey =
+        i.locationKey === undefined ? location : i.locationKey ?? undefined;
```

Comment:
What does the `?? undefined` at the end do? Does it risk negating any intentional nulls?

Reply:
after this point `null` is represented as `undefined`, so it's just converting :sweat_smile: